tests-behat/bootstrap/Context.php

```diff
@@ -275,8 +274,7 @@ public function iPressModalButton($arg)
      */
     public function modalIsOpenWithText($arg1)
     {
-        // get modal
-        $modal = $this->getSession()->getPage()->find('css', '.modal.transition.visible.active.front');
+        $modal = $this->waitForNodeElement('.modal.transition.visible.active.front');
```

Comment:
for what - with jqueryWait, we wait for AJAX + some enough time in which all animations should finish.

Reply:
Because when content has finished loading inside the modal, the modal will resize according to new content.
If you try to get the modal window while it is reloading then the test fails because the modal return null.
You can look at the commit history to see.

Reply:
I agree.

Now how to handle it - jquery wait should do that FOR EVERYTHING :)

do you interact with Modal as the "1st operation in Behat step"

Reply:
I do not see anything that will let us know that the modal window is resizing and has finish resizing... Do you?
That is why I put a loop there. The other solution will be to intentionally wait for a few ms but I think the loop is better.

Reply:
jquery wait waits some extra time for exactly these situations :)

Reply:
Well I guess it does not wait long enough... I can try to increase time there but it will increase for every step, no?

Reply:
> do you interact with Modal as the "1st operation in Behat step"

if yes, then there is probably some other issue, as the additional delay is already quite long. And I tested it very extensively - there is probably other issue.

Reply:
There is an issue and this is how to fix it...